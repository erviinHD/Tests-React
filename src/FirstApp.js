//import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

// Funcional components

const FirstApp = ({words, subtitle}) => {

    //const words = 'Hello brothers';

    return (
        <>
            <h1>{words}</h1>
            <p>{subtitle}</p>
        </>
    );
}

FirstApp.propTypes = {
    words: PropTypes.string.isRequired
}

FirstApp.defaultProps={
    subtitle: 'My first app'
}

export default FirstApp;