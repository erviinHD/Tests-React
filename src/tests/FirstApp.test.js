//import { render } from "@testing-library/react"
import FirstApp from "../FirstApp"

import React from 'react';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('Pruebas my FirstApp', () => {
    //test('Mostrar el mensaje hola mundo', () => {
    //    const saludo = "Hello"

    //    const { getByText } = render(<FirstApp words={saludo} />);

    //    expect(getByText(saludo)).toBeInTheDocument
    //})

    test('Debe mostrar <FirstApp /> correctamente', () => {
        const saludo = "Hello"
        const wrapper = shallow(<FirstApp words={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });


    test('Mostrar el subtitulo', () => {
        const saludo = "Hello"
        const sub = "obi wan kenoby"

        const wrapper = shallow(<FirstApp words={saludo} subtitle={sub} />);

        const textParrafo = wrapper.find('p').text();
        expect(textParrafo).toBe(sub)
    })
})