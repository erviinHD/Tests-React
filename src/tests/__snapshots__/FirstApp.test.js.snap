// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas my FirstApp Debe mostrar <FirstApp /> correctamente 1`] = `ShallowWrapper {}`;

exports[`Pruebas my FirstApp Mostrar el subtitulo 1`] = `"obi wan kenoby"`;
